ivy/functional/frontends/torch/linalg.py

```diff
@@ -27,3 +27,20 @@ def slogdet(input, *, out=None):
 @to_ivy_arrays_and_back
 def matrix_power(input, n, *, out=None):
     return ivy.matrix_power(input, n, out=out)
+
+
+@to_ivy_arrays_and_back
+def matrix_norm(input, ord="fro", dim=(-2, -1), keepdim=False, *, dtype=None, out=None):
+    if "complex" in ivy.as_ivy_dtype(input.dtype):
+        input = ivy.abs(input)
+    if dtype:
+        input = ivy.astype(input, dtype)
+    return ivy.matrix_norm(input, ord=ord, axis=dim, keepdims=keepdim, out=out)
+
+
+matrix_norm.supported_dtypes = (
```

Comment:
If the function supports all dtypes then there is no need to specify any supported or unsupported dtype. However, if it does support or unsupport some dtype then we would need to specify the smaller of the two sets as explained in the link above. Please checkout the deep dive section linked above and feel free to follow up on any of that. Thank you.

Reply:
But either way, setting it as an attribute `matrix_norm.supported_dtypes` is adviced against and was done few months ago and we have moved to decorators since. Some functions still need refactoring/updating.